gui.py

```
@@ -1,145 +1,466 @@
+            def update_ui():
+                self.project_select_screen.project_label.configure(text="Projects to Download")
+                
+                self.project_select_screen.project_checklist.make_checkbuttons(project_names)
+                if project_names:
+                    self.project_select_screen.download_button.configure(state="normal")
+                    # this makes the button clickable
+                else:
+                    self.project_select_screen.download_button.configure(state="disabled")
+                    # this makes the button unclickable if there are no projects to download
+            self.q.put(update_ui)
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)